src/dom.js

```diff
@@ -536,6 +536,17 @@ export function ancestorElementsByTag(child, tagName) {
   });
 }
 
+/**
+ * Iterate over an array-like
+ * @param {!IArrayLike<T>} iterable
+ * @param {!function(T, number)} cb
+ * @template T
+ */
+export function iterateCursor(iterable, cb) {
+  for (let i = 0; i < iterable.length; i++) {
```

Comment:
My test produced incorrect results, and I have updated it and it now shows the `iterable[i] != undefined` is fastest. I'll keep and update the function.